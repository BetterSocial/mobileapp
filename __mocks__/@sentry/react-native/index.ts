export const addBreadcrumb = jest.fn();
