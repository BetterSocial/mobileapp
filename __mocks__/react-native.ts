import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN