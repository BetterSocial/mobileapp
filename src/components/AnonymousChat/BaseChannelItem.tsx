import * as React from 'react';
import {View} from 'react-native';

import ChannelImage from './child/ChannelImage';
import ChannelPostNotificationMessage from './child/ChannelPostNotificationMessage';
import ChannelPostNotificationStats from './child/ChannelPostNotificationStats';
import ChannelTitle from './child/ChannelTitle';
import CustomPressable from '../CustomPressable';
import styles from './BaseChannelItemStyles';
import {
  BaseChannelItemProps,
  BaseChannelItemTypeProps
} from '../../../types/component/AnonymousChat/BaseChannelItem.types';

const BaseChannelItem: (props: BaseChannelItemProps) => React.ReactElement = ({
  anonPostNotificationUserInfo = null,
  block = 0,
  comments = 0,
  downvote = 0,
  isCommentExists = false,
  isMe = false,
  message = '',
  name = '',
  picture = 'https://fastly.picsum.photos/id/173/400/400.jpg?hmac=TU_DMkn7FSRRwiEpVveLvmyATg-y4hNrNKT-Cs4VQ1M',
  postMaker = null,
  postNotificationMessageText = '',
  postNotificationMessageUser = null,
  postNotificationPicture = 'https://fastly.picsum.photos/id/912/400/400.jpg?hmac=Wg3Y7jTiQxHr_NpRsTrHG58kBfZQGNeH2tCl5u2Ipr0',
  time = '12:00 PM',
  type = BaseChannelItemTypeProps.ANON_PM,
  unreadCount = 0,
  upvote = 0,
  onPress = () => {
    console.log('onPress');
  }
}) => {
  return (
    <CustomPressable onPress={onPress}>
      <View style={styles.chatContainer}>
        <ChannelImage
          type={type}
          mainPicture={picture}
          postNotificationPicture={postNotificationPicture}
          anonPostNotificationUserInfo={anonPostNotificationUserInfo}
          isCommentExists={isCommentExists}
          postMaker={postMaker}
        />

        <View style={styles.chatContentContainer}>
          <ChannelTitle
            type={type}
            name={name}
            message={message}
            time={time}
            unreadCount={unreadCount}
            isMe={isMe}
          />

          {/* Post Notification Message */}
          <View style={styles.chatContentSection}>
            <ChannelPostNotificationMessage
              type={type}
              commenterName={postNotificationMessageUser}
              message={postNotificationMessageText}
            />
          </View>

          {/* Post Stats */}
          <View style={styles.chatContentSection}>
            <ChannelPostNotificationStats
              type={type}
              block={block}
              upvote={upvote}
              downvote={downvote}
              comments={comments}
            />
          </View>
        </View>
      </View>
    </CustomPressable>
  );
};

export default BaseChannelItem;
