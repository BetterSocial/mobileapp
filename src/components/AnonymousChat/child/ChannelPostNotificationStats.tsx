import * as React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';

import Imageblock from '../../../assets/images/block.png';
import MemoIc_arrow_down_vote_on from '../../../assets/arrow/Ic_downvote_on';
import MemoIc_arrow_upvote_on from '../../../assets/arrow/Ic_upvote_on';
import MemoIc_block_inactive from '../../../assets/block/Ic_block_inactive';
import MemoIc_comment from '../../../assets/icons/Ic_comment';
import {BaseChannelItemTypeProps} from '../../../../types/component/AnonymousChat/BaseChannelItem.types';

const ChannelPostNotificationStats = ({
  upvote = 0,
  downvote = 0,
  comments = 0,
  block = 0,
  type = BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION
}) => {
  const styles = StyleSheet.create({
    descriptionContainer: {
      display: 'flex',
      flex: 1,
      flexDirection: 'row',
      justifyContent: 'space-between',
      marginRight: 45
    },
    avatarContainer: {},
    avatarNoHeight: {},
    iconContainer: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      marginEnd: 15
    },
    iconMargin: {},
    textVoteMargin: {
      marginStart: 5
    }
  });

  if (type === BaseChannelItemTypeProps.ANON_PM) return <></>;

  return (
    <View style={styles.descriptionContainer}>
      <View style={styles.iconContainer}>
        <MemoIc_arrow_upvote_on style={styles.iconMargin} width={15} height={15} />
        <Text style={styles.textVoteMargin}>{upvote}</Text>
      </View>
      <View style={styles.iconContainer}>
        <MemoIc_arrow_down_vote_on style={styles.iconMargin} width={15} height={15} />
        <Text style={styles.textVoteMargin}>{downvote}</Text>
      </View>
      <View style={styles.iconContainer}>
        <MemoIc_comment style={styles.iconMargin} width={15} height={15} />
        <Text style={styles.textVoteMargin}>{comments}</Text>
      </View>
      <View style={styles.iconContainer}>
        {Number(block) > 0 ? (
          <Image source={Imageblock} style={styles.iconMargin} width={15} height={15} />
        ) : (
          <MemoIc_block_inactive style={styles.iconMargin} width={15} height={15} />
        )}
        <Text style={styles.textVoteMargin}>{String(block)}</Text>
      </View>
    </View>
  );
};

export default ChannelPostNotificationStats;
