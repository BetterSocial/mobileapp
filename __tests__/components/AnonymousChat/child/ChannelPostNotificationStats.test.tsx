import * as React from 'react';
import {render} from '@testing-library/react-native';

import ChannelPostNotificationStats from '../../../../src/components/AnonymousChat/child/ChannelPostNotificationStats';
import {BaseChannelItemTypeProps} from '../../../../types/component/AnonymousChat/BaseChannelItem.types';

describe('TESTING ChatItemTargetText', () => {
  it('RENDER type ANON_PM should match snapshot', () => {
    const {toJSON} = render(
      <ChannelPostNotificationStats type={BaseChannelItemTypeProps.ANON_PM} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER not shown should match snapshot', () => {
    const {toJSON} = render(
      <ChannelPostNotificationStats
        type={BaseChannelItemTypeProps.ANON_POST_NOTIFICATION}
        shown={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER with block should match snapshot', () => {
    const {toJSON} = render(
      <ChannelPostNotificationStats
        type={BaseChannelItemTypeProps.ANON_POST_NOTIFICATION}
        shown={true}
        block={1}
        upvote={1}
        comments={1}
        downvote={0}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('RENDER with no block should match snapshot', () => {
    const {toJSON} = render(
      <ChannelPostNotificationStats
        type={BaseChannelItemTypeProps.ANON_POST_NOTIFICATION}
        shown={true}
        block={0}
        upvote={1}
        comments={1}
        downvote={0}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
