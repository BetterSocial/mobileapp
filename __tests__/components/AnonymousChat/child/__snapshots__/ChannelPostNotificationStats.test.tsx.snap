// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING ChatItemTargetText RENDER not shown should match snapshot 1`] = `
<View
  style={
    Object {
      "height": 0,
    }
  }
/>
`;

exports[`TESTING ChatItemTargetText RENDER type ANON_PM should match snapshot 1`] = `
<View
  style={
    Object {
      "height": 0,
    }
  }
/>
`;

exports[`TESTING ChatItemTargetText RENDER with block should match snapshot 1`] = `
<View
  style={
    Object {
      "height": 0,
    }
  }
/>
`;

exports[`TESTING ChatItemTargetText RENDER with no block should match snapshot 1`] = `
<View
  style={
    Object {
      "height": 0,
    }
  }
/>
`;
